src/main/java/org/assertj/core/api/junit/jupiter/SoftAssertionsExtension.java

```diff
@@ -88,41 +100,45 @@
   @Override
   public boolean supportsParameter(ParameterContext parameterContext, ExtensionContext extensionContext) {
     // Abort if parameter type is unsupported.
-    if (isUnsupportedParameterType(parameterContext.getParameter())) return false;
-
+    Class<? extends SoftAssertionsProvider> type = getSupportedParameterType(parameterContext.getParameter());
+    if (type == null) {
+    	return false;
+    }
```

Comment:
I would prefer to keep `isUnsupportedParameterType` and later call `Modifier.isAbstract(parameterContext.getParameter().getType()` as I think we don't need to cast `parameterContext.getParameter().getType()`  to `Class<? extends SoftAssertionsProvider>` for this call

Reply:
Done.